File: src/0110-平衡二叉树/0110.py
class Solution:
    def isBalanced(self, root: TreeNode) -> bool:
        def depth(root):
            if not root:
                return 0
            return 1 + max(depth(root.left), depth(root.right))

        if not root:
            return True
        left = root.left
        right = root.right

        return self.isBalanced(left) and self.isBalanced(right)  and abs(depth(left) - depth(right)) <= 1

File: src/0110-平衡二叉树/0110.cpp
class Solution
{
public:
    bool isBalanced(TreeNode *root)
    {
        return helper(root) != -1;
    }

    int helper(TreeNode *root)
    {
        if (!root)
            return 0;
        int left = helper(root->left);
        int right = helper(root->right);
        if (left == -1 || right == -1 || abs(left - right) > 1)
        {
            return -1;
        }
        return 1 + max(left, right);
    }
};